spanner/api/Program.cs

```diff
@@ -0,0 +1,965 @@
+                // Insert rows into the Singers table.
+                var cmd = connection.CreateInsertCommand("Singers",
+                    new SpannerParameterCollection {
+                        {"SingerId", SpannerDbType.Int64},
+                        {"FirstName", SpannerDbType.String},
+                        {"LastName", SpannerDbType.String}
+                    });
+                // Insert first row
+                cmd.Parameters["SingerId"].Value = 1;
+                cmd.Parameters["FirstName"].Value = "Summerian";
+                cmd.Parameters["LastName"].Value = " Chanter";
+                await cmd.ExecuteNonQueryAsync();
+                // Insert second row
+                cmd.Parameters["SingerId"].Value = 2;
+                cmd.Parameters["FirstName"].Value = "African";
+                cmd.Parameters["LastName"].Value = "Artist";
+                await cmd.ExecuteNonQueryAsync();
+                // Insert third row
+                cmd.Parameters["SingerId"].Value = 3;
+                cmd.Parameters["FirstName"].Value = "Indigenous";
+                cmd.Parameters["LastName"].Value = "Star";
+                await cmd.ExecuteNonQueryAsync();
+                // Insert rows into the Albums table.
+                cmd = connection.CreateInsertCommand("Albums",
+                    new SpannerParameterCollection {
+                        {"SingerId", SpannerDbType.Int64},
+                        {"AlbumId", SpannerDbType.Int64},
+                        {"AlbumTitle", SpannerDbType.String}
+                    });
+                // Insert first row.
+                cmd.Parameters["SingerId"].Value = 1;
+                cmd.Parameters["AlbumId"].Value = 1;
+                cmd.Parameters["AlbumTitle"].Value = "Good Hurrian Songs";
+                await cmd.ExecuteNonQueryAsync();
+                // Insert second row.
+                cmd.Parameters["SingerId"].Value = 1;
+                cmd.Parameters["AlbumId"].Value = 2;
+                cmd.Parameters["AlbumTitle"].Value = "Cuneiform Chorus";
+                await cmd.ExecuteNonQueryAsync();
+                // Insert third row.
+                cmd.Parameters["SingerId"].Value = 2;
+                cmd.Parameters["AlbumId"].Value = 1;
+                cmd.Parameters["AlbumTitle"].Value = "Zany Mbira Music";
+                await cmd.ExecuteNonQueryAsync();
+                // Insert fourth row.
+                cmd.Parameters["SingerId"].Value = 2;
+                cmd.Parameters["AlbumId"].Value = 2;
+                cmd.Parameters["AlbumTitle"].Value = "Ambient Sistrum Sirens";
+                await cmd.ExecuteNonQueryAsync();
+                Console.WriteLine("Inserted data.");
+                // Insert fifth row.
+                cmd.Parameters["SingerId"].Value = 2;
+                cmd.Parameters["AlbumId"].Value = 3;
+                cmd.Parameters["AlbumTitle"].Value = "Congo Carols";
+                await cmd.ExecuteNonQueryAsync();

```

Comment:
do you have a bulk insert example?
also the docs at some point should say that while you *can* do a Task.WhenAll(...multiple invocations of executequery...) it will consume extra sessions.